File: android/src/main/kotlin/live/hms/hmssdk_flutter/HMSHLSVariantExtension.kt
package live.hms.hmssdk_flutter

import live.hms.video.error.HMSException
import live.hms.video.sdk.models.HMSHLSVariant
import java.text.SimpleDateFormat

class HMSHLSVariantExtension {
    companion object{
        fun toDictionary(hmshlsVariant: HMSHLSVariant?):HashMap<String,Any>?{
            val args=HashMap<String,Any>()
            if (hmshlsVariant == null)return null
            args["hls_stream_url"] = hmshlsVariant.hlsStreamUrl?:""
            args["meeting_url"] = hmshlsVariant.meetingUrl?:""
            args["metadata"] = hmshlsVariant.metadata?:""
            args["started_at"] =
                SimpleDateFormat("yyyy-MM-dd HH:mm:ss").format(hmshlsVariant.startedAt).toString()
            return args
        }
    }

}

File: ios/Classes/Models/HMSHLSVariantExtension.swift
//
//  HMSHLSVariantExtension.swift
//  hmssdk_flutter
//
//  Created by Govind Maheshwari on 19/01/22.
//

import Foundation
import HMSSDK

class HMSHLSVariantExtension {

    static func toDictionary(_ hmshlsVariant: HMSHLSVariant) -> [String: Any] {

        var dict = [String: Any]()

        dict["meeting_url"] = hmshlsVariant.meetingURL.absoluteString

        dict["metadata"] = hmshlsVariant.metadata
        
        dict["hls_stream_url"] = hmshlsVariant.url.absoluteString
        
        if let startedAt = hmshlsVariant.startedAt {
            dict["started_at"] = "\(startedAt)"
        }

        return dict
    }

}
